iroh-embed/src/store.rs

```diff
@@ -0,0 +1,104 @@
+//! Store services to use in an iroh system.
+
+use std::path::PathBuf;
+
+use anyhow::Result;
+use iroh_one::mem_store;
+use iroh_rpc_client::Config as RpcClientConfig;
+use iroh_rpc_types::store::StoreAddr;
+use iroh_rpc_types::Addr;
+use iroh_store::Config as StoreConfig;
+use tokio::task::JoinHandle;
+
+/// A iroh store backed by an on-disk RocksDB.
+///
+/// An iroh system needs a store service for keeping local state and IPFS data.  This one
+/// uses RocksDB in a directory on disk.
+#[derive(Debug)]
+pub struct RocksStoreService {
+    task: JoinHandle<()>,
+    addr: StoreAddr,
+}
+
+impl RocksStoreService {
+    /// Starts a new iroh Store service with RocksDB storage.
+    ///
+    /// This implicitly starts a task on the tokio runtime to manage the storage node.
+    pub async fn new(path: PathBuf) -> Result<Self> {
+        let addr = Addr::new_mem();
+        let config = StoreConfig {
+            path,
+            rpc_client: RpcClientConfig {
+                gateway_addr: None,
+                p2p_addr: None,
+                store_addr: Some(addr.clone()),
+                channels: Some(1),
+            },
+            metrics: Default::default(),
```

Comment:
can you add some documentation about the state of metrics?

Reply:
currently i'm trying to deny their existence :)  I'm not sure it makes sense to describe in the public docs that they exist but are not yet exposed.

i created https://github.com/n0-computer/iroh/issues/597 to follow up on this though - i agree that we should provide ways to use them.

Reply:
👍